src/Grids/grid_utils.jl

```diff
@@ -157,6 +157,10 @@ Returns 1, which is the 'length' of a field along a reduced dimension.
 @inline node(LX::Nothing, LY, LZ::Nothing, i, j, k, grid) = tuple(ynode(LX, LY, LZ, i, j, k, grid))
 @inline node(LX::Nothing, LY::Nothing, LZ, i, j, k, grid) = tuple(znode(LX, LY, LZ, i, j, k, grid))
 
+@inline cpu_face_constructor_x(grid) = all_x_nodes(Face, adapt(CPU(), grid))[1:grid.Nx+1]
```

Comment:
What does `adapt(CPU(), grid)` do? Is this a function that we have defined?

Should we perhaps introduce our own function for this, rather than extending `adapt` ?